ldclient.go

```diff
@@ -583,13 +554,18 @@ func (client *LDClient) SecureModeHash(context ldcontext.Context) string {
 // this does not guarantee that the flags are up to date; if you need to know its status in more detail, use
 // [LDClient.GetDataSourceStatusProvider].
 //
+// Additionally, if the client was configured to be offline, this will always return true.
+//
 // If this value is false, it means the client has not yet connected to LaunchDarkly, or has permanently
 // failed. See [MakeClient] for the reasons that this could happen. In this state, feature flag evaluations
 // will always return default values-- unless you are using a database integration and feature flags had
 // already been stored in the database by a successfully connected SDK in the past. You can use
 // [LDClient.GetDataSourceStatusProvider] to get information on errors, or to wait for a successful retry.
 func (client *LDClient) Initialized() bool {
-	return client.dataSource.IsInitialized()
+	if client.offline {
```

Comment:
This logic used to work because if the client was in offline mode, then a `NullDataSource` was instantiated (which returned `true` for IsInitialized()). 

Since `dataSystem` doesn't expose `IsInitialized` - just the `DataAvailability` - we need to check that here explicitely. Which is much easier to understand, anyways.